products/views.py

```diff
@@ -489,3 +490,63 @@ def all_products(request):
     except EmptyPage:
         page_obj = page.page(page.num_pages)
     return render(request, 'user_product/all_product.html', {'productdata': page_obj})
+
+
+@login_required
+def banner_view(request):
+    """ This view is useful to show all banners """
+
+    banner_data = Banner.objects.filter(is_active=True)
+    search = request.GET.get('search')
+    if search is None:
+        search = ""
+    if search:
+        banner_data = banner_data.filter(banner_name__icontains=search)
+    page = Paginator(banner_data, 3)
+    page_number = request.GET.get('page')
+    try:
+        page_obj = page.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = page.page(1)
+    except EmptyPage:
+        page_obj = page.page(page.num_pages)
+    return render(request, 'product/banner/show_banner.html', {'banner_data': page_obj, 'search': search,
+                                                               'banner': banner_data})
+
+
+@login_required
+def add_banner(request):
+    """ This view is useful to create banner """
+
+    if request.method == "POST":
+        form_data = AddBannerForm(request.POST, request.FILES)
+        if form_data.is_valid():
+            form_data.save()
+        return redirect('banner')
+    form_data = AddBannerForm()
+    return render(request, 'product/banner/add_banner.html', {'form': form_data})
+
+
+@login_required
+def update_banner(request, pk):
+    """This view is useful to update banner"""
+
+    banner_data = Banner.objects.get(id=pk)
```

Comment:
@Yshah112 you can handle the exception if you are using get() method or else use the get_object_or_404() instead.

cc: @lalitvasoya

Reply:
done